sled-agent/types/src/instance.rs

```diff
@@ -47,43 +57,11 @@ pub struct InstanceEnsureBody {
     pub metadata: InstanceMetadata,
 }
 
-/// A request to attach a disk to an instance.
-#[derive(Clone, Debug, Serialize, Deserialize, JsonSchema)]
-pub struct InstanceDisk {
-    /// The disk's UUID.
-    pub disk_id: Uuid,
-    /// The logical slot number assigned to the disk in its database record.
-    pub slot: u8,
-    /// True if the disk is read-only.
-    pub read_only: bool,
-    /// A JSON representation of the Crucible volume construction request for
-    /// this attachment.
-    //
-    // This is marked as `NoDebug` because the VCR contains the volume's
-    // encryption keys.
-    pub vcr_json: NoDebug<String>,
-
-    /// The disk's name, used to generate the serial number for the virtual disk
-    /// exposed to the guest.
-    //
-    // TODO(#7153): Making this depend on the disk name means that a disk's ID
-    // may change if it is renamed or if a snapshot of it is used to create a
-    // new disk.
-    pub name: String,
-}
-
-/// Configures how an instance is told to try to boot.
-#[derive(Clone, Debug, Serialize, Deserialize, JsonSchema)]
-pub struct InstanceBootSettings {
-    /// Propolis should tell guest firmware to try to boot from devices in this
-    /// order.
-    pub order: Vec<Uuid>,
-}
-
-/// Describes the instance hardware.
+/// Describes sled-local configuration that a sled-agent must establish to make
+/// the instance's virtual hardware fully functional.
 #[derive(Clone, Debug, Serialize, Deserialize, JsonSchema)]
-pub struct InstanceHardware {
-    pub properties: InstanceProperties,
+pub struct InstanceSledLocalConfig {
```

Comment:
i *really* like splitting apart "externally-visible foundation for this VM" and "stuff we need to plumb for the VM to have the universe a user thinks it should". i dunno how much you intended that up front, but `InstanceHardware` being both was a little confusing.

Reply:
I think I stumbled across this half by accident (ISTR it fell out of an earlier draft where I was still trying to use some generated propolis-client types for certain things). But I'm with you--I'm really happy with how it turned out!